### PR TITLE
new(Aside): Add collapsible tab.

### PR DESCRIPTION
**Pull request by stefhatcher**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

- adds a collapsible tab that toggles the aside open/closed
- supported for before and after, defaults to before (being on the right)
- tab is on the outside to not conflict with the scrollbar

Layouts : Layout - With Both Asides Collapsible And Main Padding ⋅ Storybook 2019-11-05 14-29-43
Layouts : Aside - With Collapsible Before ⋅ Storybook 2019-11-05 14-30-06
Layouts : Aside - With Collapsible After ⋅ Storybook 2019-11-05 14-30-22

## Motivation and Context

adding support to toggle aside open/closed

## Testing

added new stories to storybook

## Screenshots

original designs included the tab on the inside, which doesn't work well for scrollable content:
Screen Shot 2019-11-05 at 11 01 00 AM

with `scrollable` and `collapsible`:

Layouts : Aside - As A Scrollable Container With Collapsible Tab ⋅ Storybook 2019-11-05 14-30-38

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: packages/layouts/src/components/Aside/private/Tab.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    key: 'lunar.common.collapse',
+  });
+
+  const labelExpand = T.phrase('Expand', null, {
+    context: 'Expand',
+    key: 'lunar.common.expand',
+  });
+
+  let Icon = expanded ? (
+    <IconChevronLeft accessibilityLabel={labelCollapse} size={ICON_SIZE} />
```
Comment (milesj) [inline packages/layouts/src/components/Aside/private/Tab.tsx]: `ExpandableIcon`?

Comment (milesj) [reply packages/layouts/src/components/Aside/private/Tab.tsx]: Need to handle LTR also.

**Comment by milesj** [comment]
How does this work in the context of a `Layout` component?

File: packages/layouts/src/components/Aside/private/Tab.tsx (hunk 1 above)
Comment (stefhatcher) [reply packages/layouts/src/components/Aside/private/Tab.tsx]: doh. forgot about `ExpandableIcon`. will update!

**Comment by stefhatcher** [comment]
@milesj updated with `DirectionalIcon` instead. also, for now, removed the animations. added a story to `Layout` to demonstrate it in a wider context!

File: packages/layouts/src/components/Aside/index.tsx [hunk 2]
```diff
@@ -39,24 +59,66 @@ export type Props = {
...
+  scrollable,
+  width,
+  onCollapseToggle,
+}: Props) {
   const [styles, cx] = useStyles(styleSheet);
+  const [expanded, setExpanded] = useState(true);
+
+  const handleCollapseToggle: React.DOMAttributes<HTMLButtonElement>['onClick'] = collapsible
+    ? () => {
+        setExpanded(!expanded);
```
Comment (milesj) [inline packages/layouts/src/components/Aside/index.tsx]: `setExpanded(prev => !prev)`

**Pull request merged**